app/models/foreign_input_set.rb

```diff
@@ -0,0 +1,45 @@
+class ForeignInputSet < ActiveRecord::Base
+  class CircularDependencyError < Foreman::Exception
```

Comment:
I think it's the case more when in dynflow actions, (because there is `Actions::Foreman` module), I haven't seen issues with this here